components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java

```diff
@@ -238,29 +238,49 @@ public OAuthConsumerAppDTO getOAuthApplicationDataByAppName(String appName) thro
      * Registers an OAuth consumer application.
      *
      * @param application <code>OAuthConsumerAppDTO</code> with application information
-     * @throws Exception Error when persisting the application information to the persistence store
+     * @throws IdentityOAuthAdminException Error when persisting the application information to the persistence store.
      */
     public void registerOAuthApplicationData(OAuthConsumerAppDTO application) throws IdentityOAuthAdminException {
 
+        registerAndRetrieveOAuthApplicationData(application);
+    }
+
+    /**
+     * Registers an OAuth consumer application and retrieve application details.
+     *
+     * @param application <code>OAuthConsumerAppDTO</code> with application information.
+     * @return OAuthConsumerAppDTO Created OAuth application details.
+     * @throws IdentityOAuthAdminException Error when persisting the application information to the persistence store.
+     */
+    public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumerAppDTO application)
+            throws IdentityOAuthAdminException {
+
         String tenantAwareUser = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        OAuthConsumerAppDTO oAuthConsumerAppDTO = new OAuthConsumerAppDTO();
```

Comment:
Add new util method to generate oAuthConsumerAppDTO

Reply:
Done